### PR TITLE
- Added NetworkError as a type that gets returned as part of NetworkR…

### PR DESCRIPTION
**Pull request by devgabrielcoman**
…esponse, so now we can have more descriptive erorrs *and* also get to properly handle status >= 400 type network errors

### CODE REVIEW

File: Example/SAMobileBase_Tests/Network/Task/NetworkTask_Tests.swift [hunk 1]
```diff
@@ -61,8 +89,10 @@ class NetworkTask_Tests: XCTestCase {
             expect(response).toNot(beNil())
             expect(response.status).to(equal(400))
             expect(response.error).toNot(beNil())
+            expect(response.error?.code).to(equal(400))
+            expect(response.error?.message).toNot(beNil())
+            expect(response.error?.message).to(equal("Invalid NetworkRequest object"))
```
Comment (Tbruni85) [inline Example/SAMobileBase_Tests/Network/Task/NetworkTask_Tests.swift]: I would add:

- test for no internet connection
- a correct response and be sure Network is null

**Comment by Tbruni85** [comment]
👍

**Comment by GuilhermeMota93** [comment]
👍

**Comment by Tbruni85** [comment]
👍

**Pull request merged**